llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -744,7 +744,10 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
 #include "llvm/Passes/TargetPassRegistry.inc"
 
   PB.registerPipelineStartEPCallback(
-      [](ModulePassManager &PM, OptimizationLevel Level) {
+      [this](ModulePassManager &PM, OptimizationLevel Level) {
+        PM.addPass(AMDGPUExpandPseudoIntrinsicsPass(*this));
+        FunctionPassManager FPM;
```

Comment:
Why do we need to get it run via module pass first, and then function pass again?

Reply:
That was a typo, mashed up merge, apologies.